### DISCUSSION TITLE
Angular 16 migration breaks templateRef

### DISCUSSION THREAD

**Discussion by smarsh-tejaskatariya** [l-lin/angular-datatables#1786] [Q&A] [answered]
After migration to angular 16, templateRef breaks, all the custom elements are rendered below table
<img width="927" alt="image" src="https://github.com/l-lin/angular-datatables/assets/163858824/4cda6821-0ed1-4cce-8e9e-c41c6337f93a">

It was working fine till angular 15

below are the versions of data-tables
"angular-datatables": "^16.0.0",
"datatables.net": "~1.13.4",
"datatables.net-bs5": "~1.13.4",
"datatables.net-dt": "~1.13.4",
"datatables.net-fixedcolumns": "~4.3.0",
"jquery": "^3.7.1",
"@types/datatables.net": "~1.10.28",

**Comment by shanmukhateja** [comment]
Please provide reproducible repo or StackBlitz link to investigate.

**Reply by tejaskatariya** [reply]
Let me try to create a reproducible stackblitz, till then is there anything I need to care while migrating to angular 16?

**Comment by smarsh-tejaskatariya** [comment, marked as answer]
This was breaking due to setting "preserveWhitespaces" in main.ts file.

changing 

platformBrowserDynamic()
    .bootstrapModule(AppMainModule, { preserveWhitespaces: true })
    .then(success => console.log(`Application started`))
    .catch(err => console.error(err));

to 

platformBrowserDynamic()
    .bootstrapModule(AppMainModule)
    .then(success => console.log(`Application started`))
    .catch(err => console.error(err));

solved the issue
